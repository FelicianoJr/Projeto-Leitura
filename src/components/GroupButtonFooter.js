import React from "react";
import { Link } from "react-router-dom";

const GroupButtonFooter = ({ newComment, post }) => {
  return (
    <div>
      <div className="card-footer  ">
        <Link
          className="btn btn-sm btn-light"
          to={`/${post.category}/${post.id}`}
        >
          Visualisar Comentários{" "}
          <span className="badge badge-info">{post.commentCount}</span>
        </Link>
        <button className="btn btn-sm btn-light" onClick={newComment}>
          Adicionar Comentário
        </button>
      </div>
    </div>
  );
};

export default GroupButtonFooter;
