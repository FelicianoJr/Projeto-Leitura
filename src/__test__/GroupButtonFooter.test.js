import React from "react";
import { shallow } from "enzyme";
import GroupButtonFooter from "../components/GroupButtonFooter";
import { Link } from "react-router-dom";

const setup = (post = {}) => {
  const action = {
    newComment: jest.fn()
  };
  const component = shallow(<GroupButtonFooter {...action} post={post} />);

  return {
    component: component,
    action: action,
    button: component.find("button"),
    link: component.find(Link),
    span:component.find("span")
  };
};

describe("<GroupButtonFooter/>", () => {
  let post;
  beforeEach(() => {
    post = {
      id: "rer455",
      category: "react",
      commentCount: 2
    };
  });

  it("should rende  Button", () => {
    const { button } = setup(post);
    expect(button.type()).toEqual("button");
  });

  it("should rende Link", () => {
    const { link } = setup(post);
    expect(link.props().to).toEqual("/react/rer455");
  });

  it("should rende span count 2",()=>{
    const {span} = setup(post);
    expect(span.text()).toEqual("2")
  })

  it("should click newComment ", () => {
    const { button, action } = setup(post);
    button.simulate("click");
    expect(action.newComment).toBeCalled();
  });

  it("should component Snapshot", () => {
    const { component } = setup(post);
    expect(component).toMatchSnapshot();
  });
});
