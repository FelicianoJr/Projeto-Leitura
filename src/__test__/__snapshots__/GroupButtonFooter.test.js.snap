// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupButtonFooter/> should component Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GroupButtonFooter
    newComment={[Function]}
    post={
        Object {
            "category": "react",
            "commentCount": 2,
            "id": "rer455",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card-footer  "
>
        <Link
                className="btn btn-sm btn-light"
                replace={false}
                to="/react/rer455"
        >
                Visualisar Comentários
                 
                <span
                        className="badge badge-info"
                >
                        2
                </span>
        </Link>
        <button
                className="btn btn-sm btn-light"
                onClick={[Function]}
        >
                Adicionar Comentário
        </button>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="btn btn-sm btn-light"
            replace={false}
            to="/react/rer455"
>
            Visualisar Comentários
             
            <span
                        className="badge badge-info"
            >
                        2
            </span>
</Link>,
          <button
            className="btn btn-sm btn-light"
            onClick={[Function]}
>
            Adicionar Comentário
</button>,
        ],
        "className": "card-footer  ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Visualisar Comentários",
              " ",
              <span
                className="badge badge-info"
>
                2
</span>,
            ],
            "className": "btn btn-sm btn-light",
            "replace": false,
            "to": "/react/rer455",
          },
          "ref": null,
          "rendered": Array [
            "Visualisar Comentários",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 2,
                "className": "badge badge-info",
              },
              "ref": null,
              "rendered": 2,
              "type": "span",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Adicionar Comentário",
            "className": "btn btn-sm btn-light",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Adicionar Comentário",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-footer  "
>
          <Link
                    className="btn btn-sm btn-light"
                    replace={false}
                    to="/react/rer455"
          >
                    Visualisar Comentários
                     
                    <span
                              className="badge badge-info"
                    >
                              2
                    </span>
          </Link>
          <button
                    className="btn btn-sm btn-light"
                    onClick={[Function]}
          >
                    Adicionar Comentário
          </button>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="btn btn-sm btn-light"
              replace={false}
              to="/react/rer455"
>
              Visualisar Comentários
               
              <span
                            className="badge badge-info"
              >
                            2
              </span>
</Link>,
            <button
              className="btn btn-sm btn-light"
              onClick={[Function]}
>
              Adicionar Comentário
</button>,
          ],
          "className": "card-footer  ",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Visualisar Comentários",
                " ",
                <span
                  className="badge badge-info"
>
                  2
</span>,
              ],
              "className": "btn btn-sm btn-light",
              "replace": false,
              "to": "/react/rer455",
            },
            "ref": null,
            "rendered": Array [
              "Visualisar Comentários",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 2,
                  "className": "badge badge-info",
                },
                "ref": null,
                "rendered": 2,
                "type": "span",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Adicionar Comentário",
              "className": "btn btn-sm btn-light",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Adicionar Comentário",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
